ui/tests/unit/services/api-test.js

```diff
@@ -183,4 +169,63 @@ module('Unit | Service | api', function (hooks) {
       'All supported headers are set'
     );
   });
+
+  module('Error parsing', function (hooks) {
+    hooks.beforeEach(function () {
+      this.response = {
+        errors: ['first error', 'second error'],
+        message: 'there were some errors',
+      };
+      this.getErrorResponse = () =>
+        new ResponseError({
+          status: 404,
+          url: `${document.location.origin}/v1/test/error/parsing`,
+          json: () => Promise.resolve(this.response),
+        });
+    });
+
+    test('it should correctly parse message from error', async function (assert) {
+      let e = await this.apiService.parseError(this.getErrorResponse());
+      assert.strictEqual(e.message, 'first error, second error', 'Builds message from errors');
+
+      this.response.errors = [];
+      e = await this.apiService.parseError(this.getErrorResponse());
+      assert.strictEqual(e.message, 'there were some errors', 'Returns message when errors are empty');
+
+      const error = new Error('some js type error');
+      e = await this.apiService.parseError(error);
+      assert.strictEqual(e.message, error.message, 'Returns message from generic Error');
+
+      e = await this.apiService.parseError('some random error');
+      assert.strictEqual(e.message, 'An error occurred, please try again', 'Returns default fallback');
+
+      const fallback = 'Everything is broken, sorry';
+      e = await this.apiService.parseError('some random error', fallback);
+      assert.strictEqual(e.message, fallback, 'Returns custom fallback');
+    });
```

Comment:
I consolidated all the tests from `api-error-message-test.js` into a single one here to test the various message state parsing.